### PR TITLE
Fixes #7 Refactoring the criterion tests to libcheck

### LINKED ISSUE #7: Refactor in tests
Define an interface implemented by [check](https://github.com/libcheck/check/), [criterion](https://github.com/Snaipe/Criterion) or any other you want.
This is related to the need of use the same tests battery in [libskycoin](https://github.com/skycoin/libskycoin) and hardware [wallet](https://github.com/skycoin/hardware-wallet/tree/master/skycoin-api).

### PR DESCRIPTION
**Pull request by stdevHan**
Fixes #7 

Changes:
- Refactoring the criterion tests to libcheck

Does this change need to mentioned in CHANGELOG.md?
yes

Requires testing
no

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+- Add function `SKY_base58_Hex2Base58`
```
Comment (olemis) [inline CHANGELOG.md]: Move under `Added` section

File: Makefile [hunk 2]
```diff
@@ -133,13 +134,24 @@ install-linters: ## Install linters
 
 install-deps-libc: configure-build ## Install locally dependencies for testing libskycoin
 	git clone --recursive https://github.com/skycoin/Criterion $(BUILD_DIR)/usr/tmp/Criterion
-	mkdir $(BUILD_DIR)/usr/tmp/Criterion/build
+	mkdir -p $(BUILD_DIR)/usr/tmp/Criterion/build
```
Comment (olemis) [inline Makefile]: wasn't this removed already ? isn't it using `check` now ?

File: Makefile [hunk 3]
```diff
@@ -133,13 +134,24 @@ install-linters: ## Install linters
...
 	cd    $(BUILD_DIR)/usr/tmp/Criterion/build && cmake .. && cmake --build .
 	mv    $(BUILD_DIR)/usr/tmp/Criterion/build/libcriterion.* $(BUILD_DIR)/usr/lib/
 	cp -R $(BUILD_DIR)/usr/tmp/Criterion/include/* $(BUILD_DIR)/usr/include/
 
+install-googletest-libc: configure-build ##Install googletest in debian && ubuntu
```
Comment (olemis) [inline Makefile]: why ? isn't it using `check` now ? why installing Google Test ?

File: ci-scripts/install-travis-gcc.sh [hunk 4]
```diff
@@ -1,8 +1,10 @@
+#!/usr/bin/env bash
 
 # Install gcc6 (6.4.0-2 on Mac OS) for Travis builds
 
 if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
-  sudo apt-get install -qq g++-6 && sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-6 90;
+  sudo apt-get install -qq gcc-6 g++-6
```
Comment (olemis) [inline ci-scripts/install-travis-gcc.sh]: Why did you discard `update-alternatives` ?

File: ci-scripts/install-travis-gcc.sh [hunk 5]
```diff
@@ -1,8 +1,10 @@
...
+  sudo apt-get install check
```
Comment (olemis) [inline ci-scripts/install-travis-gcc.sh]: Check is not `gcc` .

File: ci-scripts/install-travis-gcc.sh [hunk 6]
```diff
@@ -15,6 +17,8 @@ if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
   git show 42d31bba7772fb01f9ba442d9ee98b33a6e7a055:Formula/gcc\@6.rb | grep -v 'fails_with' > Formula/gcc\@6.rb
   echo 'Installing gcc@6 (6.4.0-2)'
   brew install gcc\@6 || brew link --overwrite gcc\@6
+  echo 'Installing check'
+  brew install check
```
Comment (olemis) [inline ci-scripts/install-travis-gcc.sh]: Check is not `gcc` .

File: Makefile (hunk 2 above)
Comment (stdevHan) [reply Makefile]: That is already fixed in `simelo:stdevHan_t11_device_build_matrix`

File: Makefile (hunk 3 above)
Comment (stdevHan) [reply Makefile]: That is already fixed in `simelo:stdevHan_t11_device_build_matrix`

File: ci-scripts/install-travis-gcc.sh (hunk 4 above)
Comment (stdevHan) [reply ci-scripts/install-travis-gcc.sh]: is giving me problems with the change of architectures

File: CHANGELOG.md (hunk 1 above)
Comment (stdevHan) [reply CHANGELOG.md]: ok

File: ci-scripts/install-travis-gcc.sh (hunk 5 above)
Comment (stdevHan) [reply ci-scripts/install-travis-gcc.sh]: One of the tests executed in Travis CI is called `CC=gcc-6 make test-libc`

File: ci-scripts/install-travis-gcc.sh (hunk 6 above)
Comment (stdevHan) [reply ci-scripts/install-travis-gcc.sh]: One of the tests executed in Travis CI is called `CC=gcc-6 make test-libc`

File: include/cipher.testsuite.testsuite.go.h [hunk 7]
```diff
@@ -17,7 +15,7 @@
 #define JSON_FILE_SIZE      4096
 #define JSON_BIG_FILE_SIZE  32768
 
-// FIXME: Does not work on e.g. Windowa
+// FIXME: Does not work on e.g. Windows
```
Comment (olemis) [inline include/cipher.testsuite.testsuite.go.h]: For the moment this will be tolerated ... full Windows support deferred until `0.26.0` .

File: lib/cgo/cipher.crypto.go [hunk 8]
```diff
@@ -43,36 +43,6 @@ func SKY_cipher_PubKeyFromHex(_s string, _arg1 *C.cipher__PubKey) (____error_cod
 	return
 }
 
-//export SKY_cipher_PubKeyFromSecKey
```
Comment (olemis) [inline lib/cgo/cipher.crypto.go]: why is this removed ?

File: lib/cgo/cipher.crypto.go [hunk 9]
```diff
@@ -229,18 +176,6 @@ func SKY_cipher_GenerateKeyPair(_arg0 *C.cipher__PubKey, _arg1 *C.cipher__SecKey
 	return
 }
 
-//export SKY_cipher_GenerateDeterministicKeyPair
```
Comment (olemis) [inline lib/cgo/cipher.crypto.go]: @stdevHan why removed ?

File: lib/cgo/cipher.hash.go [hunk 10]
```diff
@@ -59,14 +61,6 @@ func SKY_cipher_SHA256_Xor(_g *C.cipher__SHA256, _b *C.cipher__SHA256, _arg1 *C.
 	return
 }
 
-//export SKY_cipher_SumSHA256
-func SKY_cipher_SumSHA256(_b []byte, _arg1 *C.cipher__SHA256) (____error_code uint32) {
```
Comment (olemis) [inline lib/cgo/cipher.hash.go]: why was this removed ?

File: lib/cgo/cipher.crypto.go (hunk 9 above)
Comment (stdevHan) [reply lib/cgo/cipher.crypto.go]: Because move to https://github.com/simelo/libskycoin/blob/27e9c743fae2b479ddfeb5ed6dac252e3ddc31dd/lib/cgo/cipher.crypto.hw.go#L63

File: lib/cgo/cipher.crypto.go (hunk 8 above)
Comment (stdevHan) [reply lib/cgo/cipher.crypto.go]: Because to move https://github.com/simelo/libskycoin/blob/stdevAlDen_t7_refactor_in_tests/lib/cgo/cipher.crypto.hw.go#L139

File: lib/cgo/cipher.hash.go (hunk 10 above)
Comment (stdevHan) [reply lib/cgo/cipher.hash.go]: Because to move https://github.com/simelo/libskycoin/blob/stdevAlDen_t7_refactor_in_tests/lib/cgo/cipher.crypto.hw.go#L75

File: lib/cgo/cipher.crypto.go (hunk 8 above)
Comment (olemis) [reply lib/cgo/cipher.crypto.go]: I see ...

File: lib/cgo/libsky_handle.go [hunk 11]
```diff
@@ -467,39 +339,34 @@ func SKY_handle_close(handle C.Handle) {
 	closeHandle(Handle(handle))
 }
 
-//export SKY_handle_copy
```
Comment (olemis) [inline lib/cgo/libsky_handle.go]: can't remove this ... If I recall correctly it is used by SWIG extensions .

File: lib/cgo/cipher.crypto.go (hunk 8 above)
Comment (olemis) [reply lib/cgo/cipher.crypto.go]: Revert that change . That makes it impossible to auto-generate code out of `skycoin/skycoin` function signatures.

File: lib/cgo/cipher.crypto.hw.go [hunk 12]
```diff
@@ -0,0 +1,150 @@
+package main
```
Comment (olemis) [inline lib/cgo/cipher.crypto.hw.go]: Put these back to their original location . This makes it impossible to auto-generate code out of skycoin/skycoin function signatures. Find another way to get this done .
